js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx

```diff
@@ -43,6 +44,18 @@ import './blueprint.css';
 // when you copy-paste so they don't get pasted into editors, etc.
 patchCopyToRemoveZeroWidthUnderscores();
 
+const idempotencyLink = new ApolloLink((operation, forward) => {
+  if (/^\s*mutation/.test(operation.query.loc?.source.body ?? '')) {
```

Comment:
will defer to other reviewers about whether this is the best way to identify mutations - seems reasonable to me

Reply:
Yeah this seems like the best way for now (reading the body of the operation and testing if starts with `mutation XXX`.

At some point down the line we might start using persisted queries at which point we will need another approach to determine if a query is a mutation.

Reply:
This seems fine to me, I had thought it would be exposed in here in a more formal way but I checked the docs and it doesn't seem like it